### PR TITLE
Pneumonia Detection using X-Ray

### PR DESCRIPTION
**Pull request by PiyushBL45t**
# Pull Request for DL-Simplified 💡

## Issue Title : Pneumonia Detection using X-Ray Images
- **Info about the related issue (Aim of the project)** : Aim of the project is to detect the Pneumonia in a patient using X-Ray images
- **Name: Piyush Bhujbal** 
- **GitHub ID:** https://www.github.com/PiyushBL45t
- **Email ID:** bhujbalpiyush@gmail.com
- **Idenitfy yourself: (Mention in which program you are contributing in. Eg. For a JWOC 2022 participant it's, `JWOC Participant`)** Social Summer of Code Participant

Closes: #16 

### Describe the add-ons or changes you've made 📃
The project had a prime aim to build a machine learning model which will detect the pneumonia in the given input image of the patient's X-Ray images.

## Type of change ☑️

What sort of change have you made:

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Code style update (formatting, local variables)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

### DL Algorithms used
Developed three models for the implementation of the project. Algorithms used were:
1. EfficientNEt50
2. MobileNetV2
3. InceptionV3
4. Convolutional Neural Network

### Image Preprocessing and Overall project workflow
1. Performed image visualization of pneumonia images. Then by understanding the nature of the problem statement used the TensorFlow's image data generator to create image data for training, testing and validation.
2. Also created the NumPy arrays of images and their labels for training and testing purposes.
3. Calculated the training time for each model to get trained.
4. Trained each model for 10 epochs and plotted the loss and accuracy graphs per epoch.
5. Finally performed predictions using the testing images using OpenCV.
6. Recorded the results into the Jupyter Notebooks.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — APPROVED** [review]
Your PR is approved and ready to be merged.
@PiyushBL45t
